integration/spark/src/main/scala/org/apache/carbondata/geo/InPolygonUDF.scala

```diff
@@ -38,6 +43,46 @@ class InPolygonUDF extends (String => Boolean) with Serializable {
   }
 }
 
+@InterfaceAudience.Internal
+class InPolygonJoinRangeListUDF extends ((String, String) => Boolean) with Serializable {
+  override def apply(geoId: String, polygonRanges: String): Boolean = {
+    if (polygonRanges == null || polygonRanges.equalsIgnoreCase("null")) {
+      return false
+    }
+    // parser and get the range list
+    var range: String = polygonRanges
+    val pattern = Pattern.compile(GeoConstants.RANGELIST_REG_EXPRESSION, Pattern.CASE_INSENSITIVE)
+    val matcher = pattern.matcher(polygonRanges)
+    while ( { matcher.find }) {
+      val matchedStr = matcher.group
+      range = matchedStr
+    }
+    val ranges = PolygonRangeListExpression.getRangeListFromString(range)
```

Comment:
here we need to check against `null` for ranges? some places we check and some places we don't. (example `ToRangeListAsStringUDF`) can we make it uniform? If not required, remove from other places also. also maybe extract a common method for matcher and find if possible.

Reply:
NULL check is already handled here in line:49. Handled some refactoring to extract common code to new method.